x-pack/filebeat/module/kubernetes/envoy/ingest/pipeline.json

```diff
@@ -0,0 +1,25 @@
+{
+  "description": "Pipeline for normalizing Kubernetes envoy logs",
+  "processors": [
+    {
+      "script": {
+        "lang": "painless",
+        "source": "ctx.event.created = ctx['@timestamp']; ctx['@timestamp'] = ctx['envoy']['timestamp']; ctx.envoy.remove('timestamp');",
+        "ignore_failure" : true
+      }
+    },
+    {
+      "set": {
+        "field": "user_agent.original",
+        "value": "{{envoy.http_user_agent}}",
```

Comment:
You don't need to copy it around to avoid clobbering anymore. Since 7.0, the UA parser re-outputs the `.original`.

So you can directly populate `user_agent.original` and have:

```JSON
    {
      "user_agent": {
        "field": "user_agent.original",
        "ignore_missing": true
      }
    }
```

and the right thing will happen :-)